addons/hr_timesheet/models/project_task.py

```diff
@@ -156,14 +156,24 @@ def _get_groups(self):
 
     def action_view_subtask_timesheet(self):
         self.ensure_one()
+        is_internal_user = self.env.user.has_group('base.group_user')
         task_ids = self.with_context(active_test=False)._get_subtask_ids_per_task_id().get(self.id, [])
         action = self.env["ir.actions.actions"]._for_xml_id("hr_timesheet.timesheet_action_all")
         graph_view_id = self.env.ref("hr_timesheet.view_hr_timesheet_line_graph_by_employee").id
         new_views = []
-        for view in action['views']:
-            if view[1] == 'graph':
-                view = (graph_view_id, 'graph')
-            new_views.insert(0, view) if view[1] == 'tree' else new_views.append(view)
+        if not is_internal_user:
+            for view in action['views']:
+                if view[1] == 'tree':
+                    new_views.append((self.env.ref("hr_timesheet.hr_timesheet_line_portal_tree").id, 'tree'))
+                elif view[1] == 'form':
+                    new_views.append((self.env.ref("hr_timesheet.timesheet_view_form_portal_user").id, 'form'))
+                elif view[1] == 'kanban':
+                    new_views.append(view)
+        else:
+            for view in action['views']:
+                if view[1] == 'graph':
+                    view = (graph_view_id, 'graph')
+                new_views.insert(0, view) if view[1] == 'tree' else new_views.append(view)
```

Comment:
Is it not the case that with the current implementation we would not have access to the graph view and the pivot view ? From the video Alexandra sent it seems that we should have access to them but with the current implementation it looks like it would skip them :thinking: 
```suggestion
        for view in action['views']:
            if not is_internal_user:
                if view[1] == 'tree':
                    new_views.append((self.env.ref("hr_timesheet.hr_timesheet_line_portal_tree").id, 'tree'))
                    continue
                elif view[1] == 'form':
                    new_views.append((self.env.ref("hr_timesheet.timesheet_view_form_portal_user").id, 'form'))
                    continue
            if view[1] == 'graph':
                    view = (graph_view_id, 'graph')
            new_views.insert(0, view) if view[1] == 'tree' else new_views.append(view)
```
perhaps something like that works ? I haven't tested it just a suggestion